components/commandprovider.py

```diff
@@ -21,5 +21,5 @@ def _update_config(self, additional_config):
         self.debuglog = partial(self.logger.log, level=LogLevel.Debug)
 
     def run(self, args, shell=False, clean_return=True):
-        _run(args, shell=shell, clean_return=clean_return,
+        return _run(args, shell=shell, clean_return=clean_return,
```

Comment:
Should we go back (in a different issue) and add return code checking/handling to other places that `self.run` is used?

Reply:
That's what clean_return is intended to do: check the return code by default to make sure the command ran successfully

Reply:
Oh! I see now, thanks!